docs/core/examples/simpleclient.py

```diff
@@ -5,23 +5,39 @@
 """
 An example client. Run simpleserv.py first before running this.
 """
+from zope.interface import implementer
 
 from twisted.internet import reactor, protocol
+from twisted.internet.interfaces import IProtocol
 
 
 # a client protocol
 
 
-class EchoClient(protocol.Protocol):
+@implementer(IProtocol)
+class EchoClient:
     """Once connected, send a message, then print the result."""
 
+    def makeConnection(self, transport):
+        """
+        Called when a new connection is made.
+
+        C{transport} is used to send data to the remote peer.
+        """
+        self._transport = transport
+        print(f"New connection from remote peer {transport}")
+        self._transport.write(b"hello, world!")
```

Comment:
According to the docs for `twisted.internet.protocol.BaseProtocol`, this is supposed to call `connectionMade()`.